src/bitcaster/models/occurrence.py

```diff
@@ -84,13 +84,15 @@ def process(self) -> bool:
         notification: "Notification"
         delivered = self.data.get("delivered", [])
         recipients = self.data.get("recipients", [])
-        channels = self.event.channels.active()
+        channels = self.event.channels.active()  # List of active channel for the event related to the occurrence
         extra_filter = {}
         if limit := self.options.get("limit_to", []):
             extra_filter = {"address__value__in": limit}
+        # If context doesn't contain any filter, all the notifications/subscriptions are matched
         for notification in self.event.notifications.match(self.context):
             context = notification.get_context(self.get_context())
             for channel in channels:
+                # Here with "pending" we mean "active"
```

Comment:
this is not correct. We mean 'active' the first time, if an error occurs, this code is executed again for the remaining (not delivered) subsctiptions